### PR TITLE
Update WebApps swagger - Antares 62.1

### PR DESCRIPTION
**Pull request by nking-1**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-web/2016-03-01/swagger/Provider.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/providers/Microsoft.Web/availableStacks": {
+      "get": {
+        "tags": [
+          "Provider"
+        ],
+        "operationId": "Provider_GetAvailableStacks",
```
Comment (jianghaolu) [inline arm-web/2016-03-01/swagger/Provider.json]: Please provide a summary / description.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 2]
```diff
@@ -732,6 +732,49 @@
...
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "List the configurations of an app",
+        "description": "List the configurations of an app",
+        "operationId": "WebApps_ListConfigurations",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide `x-ms-examples` section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 3]
```diff
@@ -4289,8 +4332,50 @@
...
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/resourceHealthMetadata": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Gets the category of ResourceHealthMetadata to use for the given site",
+        "description": "Gets the category of ResourceHealthMetadata to use for the given site",
+        "operationId": "WebApps_GetResourceHealthMetadata",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide x-ms-examples section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 4]
```diff
@@ -5115,6 +5200,56 @@
...
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/config": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "List the configurations of an app",
+        "description": "List the configurations of an app",
+        "operationId": "WebApps_ListConfigurationsSlot",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide x-ms-examples section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 5]
```diff
@@ -8931,8 +9066,57 @@
...
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/resourceHealthMetadata": {
+      "get": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "Gets the category of ResourceHealthMetadata to use for the given site",
+        "description": "Gets the category of ResourceHealthMetadata to use for the given site",
+        "operationId": "WebApps_GetResourceHealthMetadataSlot",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide x-ms-examples section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 6]
```diff
@@ -9446,6 +9630,46 @@
...
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/syncfunctiontriggers": {
+      "post": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "this is to allow calling via powershell and ARM template",
+        "description": "this is to allow calling via powershell and ARM template",
+        "operationId": "WebApps_SyncFunctionTriggersSlot",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide x-ms-examples section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 7]
```diff
@@ -9446,6 +9630,46 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/syncfunctiontriggers": {
+      "post": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "this is to allow calling via powershell and ARM template",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide a more descriptive summary / description.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 8]
```diff
@@ -10405,6 +10629,39 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/syncfunctiontriggers": {
+      "post": {
+        "tags": [
+          "WebApps"
+        ],
+        "summary": "this is to allow calling via powershell and ARM template",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide a more descriptive summary / description.

File: arm-web/2016-08-01/swagger/WebApps.json [hunk 9]
```diff
@@ -10405,6 +10629,39 @@
...
+        "description": "this is to allow calling via powershell and ARM template",
+        "operationId": "WebApps_SyncFunctionTriggers",
```
Comment (jianghaolu) [inline arm-web/2016-08-01/swagger/WebApps.json]: Please provide x-ms-examples section according to https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

**Review by jianghaolu — CHANGES_REQUESTED** [review]
Thanks for the Swagger changes with good quality! Just a couple nits and a few `x-ms-examples` and we are good to go!

**Comment by Nking92** [comment]
Thank you for the review @jianghaolu. Most of these should be easy fixes. However, I'm not sure what I should do about x-ms-examples. We use Swashbuckle to generate the Swagger files, and I'm afraid that if I manually added  x-ms-examples  properties, they would be overwritten later on. Do you have any advice?

**Comment by jianghaolu** [comment]
Hi @Nking92 

Thanks! I'm looking forward to the easy fixes :)

For `x-ms-examples`, if you have .NET test recordings for these routes in your RP, we have a tool (not in production yet, but few service teams have tested) that help you get `x-ms-examples` contents. 
Clone the repo through `git clone -b readrecordings git@github.com:veronicagg/openapi-validation-tools.git` and run `node scripts/readrecording.js`.
Do you think you can try this out by early next week?

**Review by jianghaolu — COMMENTED** [review]
One more nit-picking: Please add `providers.json` to `compositeWebAppClient.json`.

**Comment by jianghaolu** [comment]
Hi @Nking92 
Thanks for addressing the review feedback! Did you get a chance to look at the example generator?

**Comment by Nking92** [comment]
Hi @jianghaolu 
We'd like to delay the examples for the next update if that's acceptable. The tooling we use to generate the swagger isn't set up to handle the examples yet, and the FTE who owns that tooling is out until May 8. In the interest of getting this new API spec published to unblock other teams for //Build work, it will be best for us to delay the examples for now. Yugang Wang has more specifics if you'd like to discuss.

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Hi @Nking92 @naveedaz
We will merge this for now but the `x-ms-examples` requirement is still valid and will be recorded as an action item on Service 360, which will be part of the monthly report. Please let us know when you plan to work on them so that we can provide the tools/resources to help you.

**Pull request merged**